apps/avatax/src/instrumentations/otel.ts

```diff
@@ -0,0 +1,27 @@
+/* eslint-disable node/no-process-env */
+// Use `process.env` here to avoid broken Next.js build
+import { ATTR_SERVICE_VERSION } from "@opentelemetry/semantic-conventions";
+import { ATTR_DEPLOYMENT_ENVIRONMENT_NAME } from "@opentelemetry/semantic-conventions/incubating";
+import { createBatchSpanProcessor } from "@saleor/apps-otel/src/batch-span-processor-factory";
+import { createHttpInstrumentation } from "@saleor/apps-otel/src/http-instrumentation-factory";
+import { registerOTel } from "@vercel/otel";
+
+import pkg from "../../package.json";
+
+registerOTel({
+  serviceName: process.env.OTEL_SERVICE_NAME,
+  attributes: {
+    [ATTR_SERVICE_VERSION]: pkg.version,
+    [ATTR_DEPLOYMENT_ENVIRONMENT_NAME]: process.env.ENV,
+    "commit-sha": process.env.VERCEL_GIT_COMMIT_SHA,
+    // unset env set by @vercel/otel as it is used by our collector
```

Comment:
this is OSS so probably better to make comment less cloud-specific, explaining general behavior